nwbinspector/checks/tables.py

```diff
@@ -183,3 +183,18 @@ def check_table_values_for_dict(table: DynamicTable, nelems: int = 200):
                 if is_string_json_loadable(string=string):
                     message += " This string is also JSON loadable, so call `json.loads(...)` on the string to unpack."
                 yield InspectorMessage(message=message)
+
+
+@register_check(importance=Importance.BEST_PRACTICE_SUGGESTION, neurodata_type=DynamicTable)
+def check_col_not_nan(table: DynamicTable, nelems: int = 200):
+    """Check if all of the values in a single column of a table are NaN."""
+    for column in table.columns:
+        if not hasattr(column, "data") or isinstance(column, VectorIndex) or isinstance(column.data[0], str):
+            continue
+        subindex_selection = np.unique(np.round(np.linspace(start=0, stop=column.shape[0] - 1, num=nelems)).astype(int))
+        if np.any(~np.isnan(column[subindex_selection])):
+            continue
+        else:
+            yield InspectorMessage(
+                message=f"Column {column.name} has all NaN values. Consider removing it from the table."
+            )
```

Comment:
It's written this way in case `np.any` has an early return condition once it detects a single non-`NaN` value

Reply:
By sparse do you mean short?

Either approach is fine with me

Reply:
I guess 'sparsity' (% of the number of `False` items being passed into `all`) isn't as relevant - I just mean for any case where there is _at least_ one `False` element occurring somewhat early on in the input to `all` (as opposed to occurring later in the vector, as seen above), it will exit a few ~`ms` faster per check (which could add up to seconds when streaming over an entire dataset).

Anyway, I implemented the `all` logic here.